### PR TITLE
Adds persistence transform example to the gallery

### LINKED ISSUE #76: More examples in the gallery on simple map and mapcube manipulation

### PR DESCRIPTION
**Pull request by Telomelonia**
## PR Description
These include persistence transform maps and mapcubes and overplotting of multiple maps in the same matplotlib Axes instance.

## TODO
- [x] make dependency for mpl_animator
- [x] changelog
- [ ] update confg.py for animation
- [ ] use different way to plot rather than `peek()`

Fixes #76

### CODE REVIEW

**Comment by Telomelonia** [comment]
pre-commit.ci autofix

**Comment by Telomelonia** [comment]
pre-commit.ci autofix

**Comment by Telomelonia** [comment]
@nabobalis 
This PR is ready, and all ci has passed.

File: changelog/137.doc.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Added an example (:ref:`sphx_glr_generated_gallery_persistence_transform.py`) illustrates persistence transform using the AIA_193_CUTOUT series.
```
Comment (nabobalis) [inline changelog/137.doc.rst:1]:
```suggestion
Added an example (:ref:`sphx_glr_generated_gallery_persistence_transform.py`) illustrates a persistence transform.
```

File: examples/persistence_transform.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
+"""
+========================================
+Persistence Transform in MapSequence
+========================================
```
Comment (nabobalis) [inline examples/persistence_transform.py:4]:
```suggestion
====================================
Persistence transform for a sequence
====================================
```

File: examples/persistence_transform.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+Persistence Transform is a simple image processing technique that is useful for the visualization
```
Comment (nabobalis) [inline examples/persistence_transform.py:5]:
```suggestion

Persistence transform is a simple image processing technique that is useful for the visualization
```

File: examples/persistence_transform.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+and depiction of gradually evolving structures.
+This example illustrates persistence transform using the AIA_193_CUTOUT series.
```
Comment (nabobalis) [inline examples/persistence_transform.py:7]:
```suggestion
This example illustrates how to perform a persistence transform.
```

File: examples/persistence_transform.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+.. note::
+   This example requires ``mpl_animators`` to be installed.
```
Comment (nabobalis) [inline examples/persistence_transform.py:10]:
```suggestion
.. note::

   This example requires `mpl_animators` to be installed.
```
If this link does not work, we should add it to the config.

File: examples/persistence_transform.py [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+"""
+
+import matplotlib.cm as cm
+import matplotlib.colors as colors
+import matplotlib.pyplot as plt
```
Comment (nabobalis) [inline examples/persistence_transform.py:15]:
```suggestion
from matplotlib import cm
from matplotlib import colors
import matplotlib.pyplot as plt
```
Do these changed imports work?

File: examples/persistence_transform.py [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+# This plots the original mapseq
+for i in aiamapseq:
+    i.peek()
+
+###############################################################################
+# This plots the final mapsequence after implementing the persistence transform
+
+result_mapseq = sunpy.map.MapSequence(persistence_maps)
+result_mapseq.peek()
+plt.show()
```
Comment (nabobalis) [inline examples/persistence_transform.py:52]:
```suggestion

plt.show()
```

File: examples/persistence_transform.py [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+import matplotlib.pyplot as plt
+import numpy as np
+
+import sunpy.map
+from sunpy.data.sample import AIA_193_CUTOUT01_IMAGE, AIA_193_CUTOUT02_IMAGE, AIA_193_CUTOUT03_IMAGE
+
+###############################################################################
+# We create the MapSequence using the AIA_193_CUTOUT sample data.
+# To create a MapSequence, we can call Map directly but add in a
+# keyword to output a MapSequence instead.
```
Comment (nabobalis) [inline examples/persistence_transform.py:24]:
```suggestion
# We will create a `~sunpy.map.MapSequence` using the `sunpy` sample data.
```

File: examples/persistence_transform.py [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+
+aiamapseq = sunpy.map.Map(AIA_193_CUTOUT01_IMAGE, AIA_193_CUTOUT02_IMAGE, AIA_193_CUTOUT03_IMAGE, cube=True)
```
Comment (nabobalis) [inline examples/persistence_transform.py:26]: Does this cube keyword work?

File: examples/persistence_transform.py [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+
+###############################################################################
+# For a data set consisting of N images with intensity values I(x,y,t),
```
Comment (nabobalis) [inline examples/persistence_transform.py:29]: Maybe this should be rendered as math text

File: examples/persistence_transform.py [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+    per = np.array([aiamapseq[n].data for n in [i, i + 1]]).max(axis=0)
+    smap = sunpy.map.Map(per, map_i.meta)
+    smap.plot_settings["cmap"] = cm.get_cmap("Greys_r")
+    smap.plot_settings["norm"] = colors.LogNorm(100, smap.max())
+    persistence_maps.append(smap)
+
+###############################################################################
+# This plots the original mapseq
+for i in aiamapseq:
+    i.peek()
```
Comment (nabobalis) [inline examples/persistence_transform.py:45]: This is not how we should be plotting these.

File: examples/persistence_transform.py [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+
+###############################################################################
+# This plots the final mapsequence after implementing the persistence transform
```
Comment (nabobalis) [inline examples/persistence_transform.py:48]:
```suggestion
# Now we will plot the original sequence and the new sequence side by side so you can see the difference.
```

Hopefully we can do this.

File: examples/persistence_transform.py [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+
+result_mapseq = sunpy.map.MapSequence(persistence_maps)
+result_mapseq.peek()
```
Comment (nabobalis) [inline examples/persistence_transform.py:51]: We should not be using peek to display any images.

**Comment by nabobalis** [comment]
I think we need to change the docs/config.py so the gallery will have animations.

File: examples/persistence_transform.py (hunk 10 above)
Comment (Telomelonia) [reply examples/persistence_transform.py:29]: > I think we need to change the docs/config.py so the gallery will have animations.

Hmm ... i will go through that 👍

File: examples/persistence_transform.py (hunk 11 above)
Comment (Telomelonia) [reply examples/persistence_transform.py:45]: I thought `peek()` was a good and more straightforward function to plot :/

File: examples/persistence_transform.py (hunk 6 above)
Comment (Telomelonia) [reply examples/persistence_transform.py:17]: Yea will change if it runs

File: examples/persistence_transform.py (hunk 11 above)
Comment (nabobalis) [reply examples/persistence_transform.py:45]: Plot is what we use for examples.

File: examples/persistence_transform.py (hunk 9 above)
Comment (Telomelonia) [reply examples/persistence_transform.py:26]: Hmm ... my bad, there is no impact from the cube keyword.

**Comment by Telomelonia** [comment]
pre-commit.ci autofix

File: examples/persistence_transform.py (hunk 9 above)
Comment (nabobalis) [reply examples/persistence_transform.py]: How do we make it a sequence?

File: examples/persistence_transform.py (hunk 11 above)
Comment (Telomelonia) [reply examples/persistence_transform.py:44]: ohh I understand ... I forgot this was example 🤦

File: examples/persistence_transform.py [hunk 14]
```diff
@@ -0,0 +1,52 @@
...
+
+###############################################################################
+# For a data set consisting of N images with intensity values I(x,y,t),
+# the Persistence Map Pn is a function of several arguments,
+# namely intensity, location and time:
+# Pn(x,y,tn)=Q(I(x,y,t≤tn))
+
+persistence_maps = []
+for i, map_i in enumerate(aiamapseq[1:]):
+    per = np.array([aiamapseq[n].data for n in [i, i + 1]]).max(axis=0)
```
Comment (wtbarnes) [inline examples/persistence_transform.py:35]: Surely we don't need two loops to do this...Can this operation not be vectorized?

File: examples/persistence_transform.py [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+    smap = sunpy.map.Map(per, map_i.meta)
+    smap.plot_settings["cmap"] = cm.get_cmap("Greys_r")
+    smap.plot_settings["norm"] = colors.LogNorm(100, smap.max())
```
Comment (wtbarnes) [inline examples/persistence_transform.py:38]: These two lines should be removed and the keyword arguments can just be passed directly when calling `plot`, e.g. `aiamapseq.plot(cmap=cm.get_cmap('Greys_r'), norm=colors.LogNorm(100, smap.max())`.

**Comment by Telomelonia** [comment]
@nabobalis 
Ohh nice, thanks for the green flag ✌️

File: examples/persistence_transform.py [hunk 16]
```diff
@@ -0,0 +1,52 @@
...
+###############################################################################
+# We will create a `~sunpy.map.MapSequence` using the `sunpy` sample data.
+
+aiamapseq = sunpy.map.Map(AIA_193_CUTOUT01_IMAGE, AIA_193_CUTOUT02_IMAGE, AIA_193_CUTOUT03_IMAGE)
+
+###############################################################################
+# For a data set consisting of N images with intensity values I(x,y,t),
+# the Persistence Map Pn is a function of several arguments,
+# namely intensity, location and time:
+# Pn(x,y,tn)=Q(I(x,y,t≤tn))
```
[the remark this answers is not in the document]
Comment (wafels) [reply examples/persistence_transform.py:31]: Typically Q is the maximum function - i.e., find the maximum value for all values from 0 to t_{n}. For a reference - DOI 10.3847/0004-637X/825/1/27 .

[the remark this answers is not in the document]
Comment (Telomelonia) [reply examples/persistence_transform.py:31]: Thanks for the reference

**Comment by nabobalis** [comment]
Thanks for working on this @Telomelonia but since there has been no activity, I will be closing this. 

If you want to continue working on this, please reopen this.

**Pull request closed without merging**